deal_finder/utils/product_meta_collector.py

```diff
@@ -0,0 +1,58 @@
+from time import sleep
+from typing import TypedDict
+from utils.transformer import TransformerUtils
+from utils.url import URLUtils
+from web_browser import WebBrowser
+
+class ProductMeta(TypedDict):
+    price: float
+    is_out_of_stock: bool
+
+web_browser = WebBrowser()
+
+class ProductMetaCollector:
+    @staticmethod
+    def __get_price(css_selector: str) -> float:
+        price = web_browser.get_element_text_by_css_selector(css_selector)
+        if not price:
+            raise Exception("Price not found!")
+
+        return TransformerUtils.get_price_from_formatted_string(price)
+
+    @staticmethod
+    def get_product_meta_from_amazon() -> ProductMeta:
```

Comment:
can we combine these 2 methods too ?

Reply:
The handling is mostly be based on the type of url we are given.\
For example: A website can disable the `BUY NOW` button rather than putting up a `not in stock message` (This might be shown when we hover over the button and is not directly accessible on the website).\
We will need some specific handling in that case.

Later when we add more websites in future, the code will increase up and become complex to handle in a single function. So breaking it up seemed like a good option to go about.\
Let me know if your thoughts differ on this matter.